topic_mutes: Filter deactivated streams from get_topic_mutes.

lukem1 opened the pull request:
Updated database query to filter out deactivated streams from the return of the get_topic_mutes method.

<!-- What's this PR for?  (Just a link to an issue is fine.) -->


Fixes blueslip warnings thrown by muting.js set_muted_topics when passed deactivated streams via page_params.
Discussed during review of #17249.

**Testing plan:** <!-- How have you tested? -->

Passed automated testing and performed manual testing.


<!-- Also be sure to make clear, coherent commits:
  https://zulip.readthedocs.io/en/latest/contributing/version-control.html
  -->

timabbott commented:
@lukem1 thanks for working on this!  Can you add a test for this corner case?  This seems like the sort of detail that we'll want to make sure we don't break in future work, and the best way to arrange that is to write a test :)

lukem1 commented:
@timabbot Thanks for the feedback! I have gone ahead and implemented a test for this edge case. I also added an optional parameter to `get_topic_mutes` to provide access to the original behavior of including the deactivated streams in the query to avoid entirely removing access to muted topics in deactivated streams via this method.

I looked into the behavior of `exclude_topic_mutes` and I think the current behavior is correct, I have added a comment to clarify this as you suggested. If this method was modified to match the new behavior of `get_topic_mutes` then it would no longer return conditions to exclude muted topics that are part of deactivated streams, and I can imagine some cases where this would cause issues when using this method to exclude muted topics from a set of topics which includes those from deactivated streams.

Let me know if you have any other feedback/suggestions! Also, apologies for the delay! Just finishing up with midterm exams :)

lukem1 commented:
@zulipbot add "needs review"
